python/cudf/cudf_pandas_tests/third_party_integration_tests/dependencies.yaml

```diff
@@ -244,6 +251,15 @@ dependencies:
           - pip
           - pip:
             - xgboost>=2.0.1
+  test_catboost:
+    common:
+      - output_types: conda
+        packages:
+        # TODO: Remove numpy pinning once https://github.com/catboost/catboost/issues/2671 is resolved
```

Comment:
See this paragraph from the numpy 2 release

> Breaking changes to the NumPy ABI. As a result, binaries of packages
that use the NumPy C API and were built against a NumPy 1.xx release
will not work with NumPy 2.0. On import, such packages will see an
ImportError with a message about binary incompatibility.